src/passes/OptimizeInstructions.cpp

```diff
@@ -2225,6 +2257,24 @@ struct OptimizeInstructions
         matches(curr, binary(And, pure(&left), ival(0)))) {
       return right;
     }
+    // -x * C   ==>    x * -C,   if  shrinkLevel != 0  or  C != C_pot
+    // -x * C   ==>   -(x * C),  otherwise
+    //    where  x, C  are integers
+    Binary* inner;
+    if (matches(
+          curr,
+          binary(Mul, binary(&inner, Sub, ival(0), any(&left)), ival()))) {
+      if (getPassOptions().shrinkLevel != 0 ||
+          !Bits::isPowerOf2(right->value.getInteger())) {
```

Comment:
It would be good to have tests at multiple shrink levels.

Reply:
done!